src/batched-json.js

```diff
@@ -60,18 +60,20 @@ export function batchFetchJsonFor(
     // Throw user error if this element is performing URL substitutions
     // without the soon-to-be-required opt-in (#12498).
     if (opt_urlReplacement == UrlReplacementPolicy.OPT_IN) {
-      urlReplacements.collectUnwhitelistedVars(element).then(unwhitelisted => {
-        if (unwhitelisted.length > 0) {
+      return urlReplacements.collectUnwhitelistedVars(element).then(invalid => {
+        if (invalid.length > 0) {
           const TAG = element.tagName;
-          user().error(TAG, 'URL variable substitutions in CORS fetches ' +
-              'triggered by amp-bind will soon require opt-in. Please add ' +
-              `data-amp-replace="${unwhitelisted.join(' ')}" to the ` +
-              `<${TAG}> element. See "bit.ly/amp-var-subs" for details.`);
-          // For tracking in error reporting.
-          dev().expectedError(TAG, 'amp-bind CORS fetch without opt-in!');
+          throw user().createError('URL variable substitutions in CORS ' +
+              'fetches from dynamic URLs (e.g. via amp-bind) require opt-in. ' +
+              `Please add data-amp-replace="${invalid.join(' ')}" to ` +
+              `the <${TAG}> element. See "bit.ly/amp-var-subs" for details.`);
+        } else {
```

Comment:
Else is unnecessary.

Reply:
Won't the chained promise below `.then(src => ...` fail to receive the `src` value? This would be easier to read if sync. :)